graph/resolver/flair.resolvers.go

```diff
@@ -4,14 +4,43 @@ package resolver
 
 import (
 	"context"
-	"fmt"
 
 	"github.com/nedrocks/delphisbe/graph/generated"
 	"github.com/nedrocks/delphisbe/graph/model"
 )
 
-func (r *flairResolver) ImageURL(ctx context.Context, obj *model.Flair) (*model.URL, error) {
-	panic(fmt.Errorf("not implemented"))
+func (r *flairResolver) DisplayName(ctx context.Context, obj *model.Flair) (*string, error) {
+	if obj.Template == nil {
+		template, err := r.DAOManager.GetFlairTemplateByID(ctx, obj.TemplateID)
+		if err != nil || template == nil {
+			return nil, err
+		}
+		obj.Template = template
+	}
+	return obj.Template.DisplayName, nil
+}
+
+func (r *flairResolver) ImageURL(ctx context.Context, obj *model.Flair) (*string, error) {
+	if obj.Template == nil {
+		template, err := r.DAOManager.GetFlairTemplateByID(ctx, obj.TemplateID)
+		if err != nil || template == nil {
+			return nil, err
+		}
+		obj.Template = template
+	}
+	return obj.Template.ImageURL, nil
+}
+
+func (r *flairResolver) Source(ctx context.Context, obj *model.Flair) (string, error) {
+	if obj.Template == nil {
+		template, err := r.DAOManager.GetFlairTemplateByID(ctx, obj.TemplateID)
+		if err != nil || template == nil {
+			// TODO: Not sure what to return here... This really should never happen.
```

Comment:
If `err == nil && template == nil` let's return an error rather than an arbitrary string.

Reply:
The problem here is more that this method requires returning a String, instead of a *String. So I can't just return `nil, err` like in other places. I could just make it an optional string instead. But every flair should have a source. The only time this would ever happen is if somehow the template is deleted or ID changed and there's a DB integrity error, i.e. can't get the template.

Reply:
good point -- however we still need to return an error in this  case (currently this flow will return a nil error here. You can just return empty string and construct an error in that case though.